Tests/Marketplace/marketplace_services.py

```diff
@@ -3693,3 +3689,33 @@ def get_last_commit_from_index(service_account):
     index_string = index_blob.download_as_string()
     index_json = json.loads(index_string)
     return index_json.get('commit')
+
+
+def get_id_set_entity_by_path(entity_path: Path, pack_folder: str, id_set: dict):
+    """
+    Get the full entity dict from the id set of the entity given as a path, if it does not exist in the id set
+    return None.
+    Args:
+        entity_path: path to entity (content item)
+        pack_folder: containing folder of that item
+        id_set: id set dict
+
+    Returns:
+        id set dict entity if exists, otherwise {}
+    """
+
+    for id_set_entity in id_set[PACK_FOLDERS_TO_ID_SET_KEYS[pack_folder]]:
+
+        if len(entity_path.parts) == 5:  # Content items that have a sub folder (integrations, scripts, etc)
```

Comment:
can't we check `entity_path in list(id_set_entity.values())[0]['file_path']`

Reply:
The id-set holds only yml/json files for the content entities, so if for example a change was made in an integration python file only, then it will not be calculated because the `entity_path` is the modified python file and it compares with the integration yaml file in the id-set.

Reply:
add it as a comment

Reply:
That was the original thing causing this bug - checking if the specific path in the id set was modified or not. this means any other file that is not the yml file that was modified will be missed.